CRM/Price/BAO/LineItem.php

```diff
@@ -444,6 +444,12 @@ public static function processPriceSet($entityId, $lineItem, $contributionDetail
           if ($line['entity_table'] == 'civicrm_contribution') {
             $line['entity_id'] = $contributionDetails->id;
           }
+          // CRM-19094: entity_table is set to civicrm_membership then ensure
+          // the entityId is set to membership ID not contribution by default
+          elseif ($line['entity_table'] == 'civicrm_membership' && !empty($line['entity_id']) && $line['entity_id'] == $contributionDetails->id) {
```

Comment:
This looks like an after the fact  fixup hack rather than a solve the problem where it arises. Any luck tracing down where $line['entity_id'] is set in the flow leading to error? 

(As a very minor aside, it's conceivable a contribution to pay for a membership might have same id as membership. But the assignment below won't change the value so there is no issue.)